packages/flutter/test/material/slider_theme_test.dart

```diff
@@ -590,6 +590,142 @@ void main() {
+        overlayShape: SliderComponentShape.noOverlay,
+        thumbShape: SliderComponentShape.noThumb,
+        showValueIndicator: ShowValueIndicator.never,
+      ),
+      value: 0.5,
+      divisions: 4
+    ));
+
+    final RenderBox sliderBox = tester.firstRenderObject<RenderBox>( find.byType(Slider));
+
+    // 5 tick marks.
+    expect(sliderBox, paintsExactlyCountTimes(#drawCircle, 5));
+  });
+
+  testWidgets('The slider can skip all component painting except the thumb', (WidgetTester tester) async {
+    // Pump a slider with just a thumb.
+    await tester.pumpWidget(_buildApp(
+      ThemeData().sliderTheme.copyWith(
+        trackHeight: 0,
+        overlayShape: SliderComponentShape.noOverlay,
+        tickMarkShape: SliderTickMarkShape.noTickMark,
+        showValueIndicator: ShowValueIndicator.never,
+      ),
+      value: 0.5,
+      divisions: 4
+    ));
+
+    final RenderBox sliderBox = tester.firstRenderObject<RenderBox>(find.byType(Slider));
+
+    // 1 thumb.
+    expect(sliderBox, paintsExactlyCountTimes(#drawCircle, 1));
+  });
+
+  testWidgets('The slider can skip all component painting except the overlay', (WidgetTester tester) async {
+    // Pump a slider with just an overlay.
+    await tester.pumpWidget(_buildApp(
+      ThemeData().sliderTheme.copyWith(
+        trackHeight: 0,
+        thumbShape: SliderComponentShape.noThumb,
+        tickMarkShape: SliderTickMarkShape.noTickMark,
+        showValueIndicator: ShowValueIndicator.never,
+      ),
+      value: 0.5,
+      divisions: 4
+    ));
+
+    final RenderBox sliderBox = tester.firstRenderObject<RenderBox>(find.byType(Slider));
+
+    // Tap the center of the track and wait for animations to finish.
+    final Offset center = tester.getCenter(find.byType(Slider));
+    TestGesture gesture = await tester.startGesture(center);
+    await tester.pumpAndSettle();
+
+    // 1 overlay.
+    expect(sliderBox, paintsExactlyCountTimes(#drawCircle, 1));
+
+    await gesture.up();
+  });
+
+  testWidgets('The slider can skip all component painting except the value indicator', (WidgetTester tester) async {
+    // Pump a slider with just a value indicator.
+    await tester.pumpWidget(_buildApp(
+      ThemeData().sliderTheme.copyWith(
+        trackHeight: 0,
+        overlayShape: SliderComponentShape.noOverlay,
+        thumbShape: SliderComponentShape.noThumb,
+        tickMarkShape: SliderTickMarkShape.noTickMark,
+        showValueIndicator: ShowValueIndicator.always,
+      ),
+      value: 0.5,
+      divisions: 4
+    ));
+
+    final RenderBox sliderBox = tester.firstRenderObject<RenderBox>(find.byType(Slider));
+
+    // Tap the center of the track and wait for animations to finish.
+    final Offset center = tester.getCenter(find.byType(Slider));
+    TestGesture gesture = await tester.startGesture(center);
+    await tester.pumpAndSettle();
+
+    // 1 value indicator.
+    expect(sliderBox, paintsExactlyCountTimes(#drawPath, 1));
```

Comment:
I think we can strengthen the tests a bit by also checking that 0 circles/rects were painted. Like-wise for the other tests too

Reply:
Done